legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java

```diff
@@ -494,12 +495,16 @@ private boolean connectionFactoryIsUsingInVMConnectors(OperationContext context,
     private void migrateDiscoveryGroup(ModelNode newAddOp, List<String> warnings) {
         // These attributes are not present in the messaging-activemq subsystem.
         // Instead a socket-binding must be used to configure the broadcast-group.
+        boolean mandatorySocketBinding = false;
         for (String property : asList(LOCAL_BIND_ADDRESS.getName(), GROUP_ADDRESS.getName(), GROUP_PORT.getName())) {
             if (newAddOp.has(property)) {
                 newAddOp.remove(property);
                 warnings.add(ROOT_LOGGER.couldNotMigrateDiscoveryGroupAttribute(property, pathAddress(newAddOp.get(OP_ADDR))));
+                mandatorySocketBinding = true;
             }
         }
+        if (mandatorySocketBinding)
+            newAddOp.get(SOCKET_BINDING.getName()).set("group-x-binding");
```

Comment:
-1 the migrate operation should not add reference to unexisting socket-binding.

The correct way to migrate such a resource would be to migrate the messaging subsystem, add the socket-binding resource and then update the socket-binding attribute of the broadcast-groups resource